Mathlib/Data/Nat/PSub.lean

```diff
@@ -71,11 +71,13 @@ theorem psub_eq_some {m : ℕ} : ∀ {n k}, psub m n = some k ↔ k + n = m
     simp [add_comm, add_left_comm]
 
 theorem psub_eq_none {m n : ℕ} : psub m n = none ↔ m < n := by
-  rcases s : psub m n <;> simp [eq_comm]
-  · refine lt_of_not_ge fun h => ?_
+  rcases s : psub m n
+  · simp only [true_iff]
+    refine lt_of_not_ge fun h => ?_
     obtain ⟨k, e⟩ := le.dest h
     injection s.symm.trans (psub_eq_some.2 <| (add_comm _ _).trans e)
-  · rw [← psub_eq_some.1 s]
+  · simp only [eq_comm, reduceCtorEq, false_iff, not_lt]
```

Comment:
optional: `grind [psub_eq_some]` does this whole branch

Reply:
I like it!